libevmasm/SimplificationRule.h

```diff
@@ -36,9 +36,32 @@ namespace solidity
 template <class Pattern>
 struct SimplificationRule
 {
+	SimplificationRule(
+		Pattern _pattern,
+		std::function<Pattern()> _action,
+		bool _removesNonConstants
+	):
+		pattern(_pattern),
+		action(_action),
+		removesNonConstants(_removesNonConstants)
+	{}
+
+	SimplificationRule(
+		Pattern _pattern,
+		std::function<Pattern()> _action,
+		bool _removesNonConstants,
+		std::function<bool()> _feasible
+	):
+		pattern(_pattern),
```

Comment:
```suggestion
		pattern(std::move(_pattern)),
```

Reply:
On that note I'm also wondering, whether the rule list should use ``emplace_back(...)`` rather than ``push_back({...})`` - but that's for another PR.